components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -2514,3 +2540,323 @@ def object_table_query(request, objtype, objid, conn=None, **kwargs):
     tableData['parentId'] = ann['parentId']
     tableData['addedOn'] = ann['addedOn']
     return tableData
+
+
+def build_url(request, name, api_version, **kwargs):
```

Comment:
@will-moore `build_absolute_url` uses `get_host` that reads on `request.META` that can poison cache. By default Django introduced site framework https://docs.djangoproject.com/en/1.8/ref/contrib/sites/#how-django-uses-the-sites-framework and I think alternatively adding that to settings won't hart. I think I didn't suggest that as a default solution.